class_ex/ex2/ex2.py

```diff
@@ -1,26 +1,62 @@
 class Person:
-    def __init__(self, name: str):
-        self._name = name
+    def __init__(self, name: str, age: int):
+        if not isinstance(age, int):
+            raise ValueError("Age has to be int type")
+        if not isinstance(name, str):
+            raise ValueError("Name has to be str type")
+        self.__name = name
+        self.__age = age
 
+    @property
+    def name(self):
+        return self.__name
 
-class Count(Person):
-    def __init__(self, name: str, cant: float = 0.0):
-        super().__init__(name)
-        if type(name) is not str:
-            raise SyntaxError("Name has to be str type")
-        self._cant = cant
+    @property
+    def age(self):
+        return self.__age
 
-    def __str__(self):
-        return f"Your count has: {self._cant}$"
 
-    def get_cant(self):
-        return self._cant
+class Count:
+    def __init__(self, p1: Person, cant: float = 0.0):
+        if not isinstance(p1, Person):
+            raise ValueError("Name has to be a Person object")
+        if not isinstance(cant, float):
+            raise ValueError("Cant has to be float type")
+        self.__p1 = p1
+        self.__cant = cant
+
+    @property
+    def p1(self):
+        return self.__p1
+
+    @property
+    def cant(self):
+        return self.__cant
+
+    def __str__(self):
+        return f"{self.__p1.name}, your count has: {self.__cant}$"
 
     def add_cant(self, value: float):
-        if value >= 0:
-            self._cant += value
+        if value >= 0 and isinstance(value, float):
+            self.__cant += value
         else:
-            ValueError("The value has to be greater than 0")
+            ValueError("The value has to be a number greater than 0")
 
     def sub_cant(self, value: float):
-        self._cant = round(self._cant - value, 2)
+        if not isinstance(value, int):
```

Comment:
```suggestion
def sub_cant(self, value: int | float):
        if not isinstance(value, int | float):
```
Comprobamos que pueda usar la función con un entero o un float

Reply:
fixed